core/src/main/java/com/scalar/db/transaction/rpc/GrpcTransactionManager.java

```diff
@@ -88,24 +90,16 @@ public void withTable(String tableName) {
 
   @Override
   public GrpcTransaction start() throws TransactionException {
-    return execute(
-        () -> {
-          StartTransactionResponse response =
-              stub.start(StartTransactionRequest.newBuilder().build());
-          return new GrpcTransaction(
-              response.getTransactionId(), stub, metadataManager, namespace, tableName);
-        });
+    GrpcTransactionService service = new GrpcTransactionService(stub, metadataManager);
```

Comment:
I'm wondering if it should create `GrpcTransactionService` object in every start.
If `GrpcTransactionManager` object singleton is shared (, which is most of the cases,) it seems not really effective?
Or, should `GrpcTransactionManager` be created for each transaction?

Ah, maybe, it's created for managing `finished` and the latch?

Reply:
Thank you for reviewing this!

> Ah, probably, it's created for managing finished and the latch separately?

Yes, `GrpcTransactionService` indicates things like a transaction session (assigned to a gRPC bidirectional stream), so we need to instantiate it per transaction. Maybe, the name (GrpcTransactionService) is not good. I will try to rename it to a better one. Thanks!